test/test_oneagentmetadataenricher.py

```diff
@@ -58,10 +75,122 @@ def test_mock_get_metadata_file(self, mock_func):
+        enricher = OneAgentMetadataEnricher()
         tags = {"tag1": "value1"}
         mock_func.return_value = {"tag1": "newValue"}
 
         enricher.add_oneagent_metadata_to_tags(tags)
 
         self.assertEqual(tags, {"tag1": "newValue"})
+
+    @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
+           '.OneAgentMetadataEnricher._get_metadata_file_content')
+    def test_parse_valid_data(self, mock_func):
+        mock_func.return_value = ["k1=v1\n", "k2=v2"]
+
+        enricher = OneAgentMetadataEnricher()
+        res = enricher._parse_oneagent_metadata(
+            enricher._get_metadata_file_content())
+
+        self.assertEqual(2, len(res))
+        self.assertEqual("v1", res["k1"])
+        self.assertEqual("v2", res["k2"])
+
+    @patch('dynatrace.opentelemetry.metrics.export.oneagentmetadataenricher'
+           '.OneAgentMetadataEnricher._get_metadata_file_content')
+    def test_parse_empty_list(self, mock_func):
+        mock_func.return_value = []
+
+        enricher = OneAgentMetadataEnricher()
+        res = enricher._parse_oneagent_metadata(
+            enricher._get_metadata_file_content())
+
+        self.assertEqual(0, len(res))
+
+
+class TestGetIndirectionFile(unittest.TestCase):
+    def test_get_indirection_file_contents_success(self):
+        # when open is called, a file with this content is returned
+        mock = mock_open(read_data="metadata_file_name")
+        enricher = OneAgentMetadataEnricher()
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_name("not-used-since-mocking")
+            self.assertEqual("metadata_file_name", res)
+
+    def test_get_indirection_file_missing_or_not_readable(self):
+        # will throw an IOError when calling open
+        mock = mock_open()
+        mock.side_effect = IOError()
+        enricher = OneAgentMetadataEnricher()
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_name("not-used-since-mocking")
+            self.assertEqual(None, res)
+
+    def test_get_indirection_fname_missing_empty_or_invalid(self):
+        enricher = OneAgentMetadataEnricher()
+        self.assertEqual(None, enricher._get_metadata_file_name(""))
+        self.assertEqual(None, enricher._get_metadata_file_name(None))
+        self.assertEqual(None, enricher._get_metadata_file_name("#%&^:"))
+
+
+class TestGetMetadataContents(unittest.TestCase):
+    def test_indirection_file_empty(self):
+        mock = mock_open(read_data="")
+        enricher = OneAgentMetadataEnricher()
+
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_content()
+            self.assertEqual(0, len(res))
+
+    def test_get_file_contents_success(self):
+        mock = mock_open(read_data="key1=value1\nkey2=value2")
+        enricher = OneAgentMetadataEnricher()
+        with patch("builtins.open", mock):
+            res = enricher._get_metadata_file_content()
```

Comment:
This test is not quite valid since it will return `"key1=value1\nkey2=value2"` as file name on the first open, try to open this file name and it will succeed because the second open call is also mocked.
I think `test_get_file_contents_from_tmpfile` below, which tests it properly, is sufficient, so you can remove this misleading test case here or add a comment that this only covers `_get_metadata_file_content` partially.

Reply:
oh, that's true! updated it so it will return `indirection_filename` on the first call. also added the info that it only partially tests the function